fs/vfs/fs_fdopen.c

```diff
@@ -120,7 +120,7 @@ int fs_fdopen(int fd, int oflags, FAR struct tcb_s *tcb,
 
   /* Check input parameters */
 
-  if (fd < 0)
+  if (fd < 0 && cookie == 0)
```

Comment:
@xiaoxiang781216 This is basically just a information passed to fs_fdopen that initialization is done for fopencookie (1 true) and not standard stream (0 false). It is not to pass the pointer to cookie. The implementation with pointer to cookie would be tricky because we need valid fd to pass the checks (and we need -1 reserved if open call returns error). This adds two more if statements but does not complicate the code more. Applies to most other comments.

Reply:
with https://github.com/apache/nuttx/pull/10913, you could move FILE allocation into fopencookie, and forward fdopen to fopencookie.